docs/read-more/how-to-write-a-sequence.md

```diff
@@ -0,0 +1,251 @@
+```
+Then you can send it JSON strings separated by newline characters.  
+
+### Typescript
+Sequences that only consumes data should be typed as [WritableApp](https://hub.scramjet.org/docs/types/modules#writableapp).
+
+### Writing to input stream
+
+You can write to instance input stream using our CLI, REST API, or API Client.
+
+## Transforming data
+Transforming data is really a combination of consuming and producing, usually with some logic in between. Let’s filter the incoming input stream of numbers to include only the even ones. We will also have to consider backpressure, because we are producing data.
+
+#### Using async iteration and a generator:
+```ts
+export default async function*(input) {
+    for await (const num of input) {
+        if(num % 2 === 0) {
+            yield num
+        }
+    }
+}
+```
+
+#### Using streams:
+While using raw streams we need to handle backpressure ourselves.
+```ts
+export default function(input) {
+    const out = new PassThrough()
+
+    input.on('data', (num) => {
+        if(num % 2 === 0) {
+            const canWrite = out.write(num)
+            if(!canWrite)
+                input.pause()
+        }
+    })
+
+    out.on('drain', () => input.resume())
+
+    return out
+}
+```
+
+### Typescript
+Sequences that transform data should be typed as [TransformApp](https://hub.scramjet.org/docs/types/modules#transformapp).  
+[Here's an example](https://github.com/scramjetorg/scramjet-cloud-docs/blob/main/samples/transform-string-stream/src/index.ts).
+
+## Sequence arguments
+Every sequence can be spawned with arbitrary number of arguments
+```bash
+si seq start <sequence-id> --args ['Hello', 123, { abc: 456 }]
+```
+you can access this args using function parameters in your sequence:
+```ts
+export default function(input, param1, param2., param3) {
+    console.log(param1 + ' ' param2 + ' ' param3.abc)
+    // Prints "Hello 123 456" to stdout
+}
+```
+
+## Sending data between sequences (topics)
+Sometimes you need a bunch of sequences to talk to each other. Topics are the solution. It’s a PubSub system that allows for many writing and reading instances to exchange data. 
+Topics are either attached to the input stream or routed from the output stream.
+### Writing to a topic 
+To have your output stream routed to a topic, it needs to have a property `topic` with a topic name and `contentType` for proper consumer encoding. 
+```ts
+export default function() {
+    const out = new PassThrough()
+
+    temparatureSensor.on('update', (data) => {
+        out.write(data)
+    })
+
+    out.topic = 'temperature-readouts'
+    out.contentType = 'text/plain'
+
+    return out
+}
+```
+### Reading from a topic 
+To send topic as input stream we specify a topic config object with two properites: `requires` - with a name of topic, and `contentType` for ensuring proper encoding.
+```ts
+const CRITICAL_TEMP_CELCIUS = 40
+
+const app = [
+    {
+        requires: 'temperature-readouts',
```

Comment:
You described topics very well.
I have a question to @MichalCz about our support for defining topics within the Sequence code, because he mentioned on our last Friday's meeting we should deprecate it and only support defining topics when starting an Instance the way you have implemented it recently. So the question is whether we want to teach users to write the Sequences that way.

Reply:
Will ask @MichalCz about that

Reply:
thanks